### PR TITLE
Update to latest Smol Farm ABI

### PR DESCRIPTION
**Pull request by alecananian**
- Updates to latest Smol Farm ABI that handles staked brains and bodies
- Adds reward token result to Claims when they move to Claimed status
- Adds ERC721 transfer handlers to collect token data for Smol Bodies, Smol Brains, and Smol Brains Land
- Adds staking/unstaking handlers for Smol School and Gym

### CODE REVIEW

**Comment by alecananian** [comment]
Subgraph deploy is timing out 😢

File: packages/config/src/arbitrum-testnet.json [hunk 1]
```diff
@@ -38,28 +38,30 @@
...
+  "smol_brains_land_start_block": 9382088,
   "smol_brains_pets_address": "0x0000000000000000000000000000000000000000",
   "smol_brains_pets_start_block": 9265867,
   "smol_brains_school_address": "0x0000000000000000000000000000000000000000",
   "smol_brains_school_start_block": 9265867,
   "smol_cars_address": "0x0000000000000000000000000000000000000000",
   "smol_cars_start_block": 9265867,
-  "smol_farm_address": "0xfE8c1ac365bA6780AEc5a985D989b327C27670A1",
-  "smol_farm_start_block": 9141924,
+  "smol_farm_address": "0x5b18510C5f6a00c3b8Cd5479a0b791E8950C36E8",
```
Comment (wyze) [inline packages/config/src/arbitrum-testnet.json:51]: Does this point to the proxy or the implementation address?

File: subgraphs/smolverse/src/helpers/collections.ts [hunk 2]
```diff
@@ -3,21 +3,27 @@ import {
   SMOL_BODIES_ADDRESS,
   SMOL_BODIES_PETS_ADDRESS,
   SMOL_BRAINS_ADDRESS,
-  SMOL_BRAINS_PETS_ADDRESS
+  SMOL_BRAINS_PETS_ADDRESS,
+  SMOL_TREASURES_ADDRESS
 } from "@treasure/constants";
+import { SMOL_BRAINS_LAND_ADDRESS } from "@treasure/constants";
```
Comment (wyze) [inline subgraphs/smolverse/src/helpers/collections.ts:9]: Combine the imports here.

**Review by wyze — CHANGES_REQUESTED** [review]
One minor import correction.

File: packages/config/src/arbitrum-testnet.json (hunk 1 above)
Comment (alecananian) [reply packages/config/src/arbitrum-testnet.json:51]: This is the proxy

File: subgraphs/smolverse/src/mappings/erc721.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  const token = getOrCreateToken(collection, tokenId);
+
+  const contract = ERC721.bind(address);
+  const tokenUri = contract.try_tokenURI(tokenId);
+  token.owner = owner.id;
+
+  if (isMint(params.from)) {
+    token.name = `${collection.name} #${tokenIdString}`;
+    if (!tokenUri.reverted) {
+      // const data = ipfsData || getIpfsJson(tokenUri.value);
```
Comment (wyze) [inline subgraphs/smolverse/src/mappings/erc721.ts:27]:
```suggestion
```

Removing commented code.

File: subgraphs/smolverse/src/mappings/erc721.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  const address = event.address;
+  const tokenId = params.tokenId;
+  const tokenIdString = tokenId.toString();
+
+  const collection = getOrCreateCollection(address);
+  const owner = getOrCreateUser(params.to.toHexString());
+  const token = getOrCreateToken(collection, tokenId);
+
+  const contract = ERC721.bind(address);
+  const tokenUri = contract.try_tokenURI(tokenId);
```
Comment (wyze) [inline subgraphs/smolverse/src/mappings/erc721.ts:21]: Lets move this into the `isMint` if block. This is expensive so we only want to do it when needed.

File: subgraphs/smolverse/src/mappings/smol-pets.ts [hunk 5]
```diff
@@ -24,55 +22,18 @@ export function handleMint(event: SmolPetMint): void {
 
   // const data = ipfsData || getCollectionJson(collection, `${token.tokenId}.json`);
```
Comment (wyze) [inline subgraphs/smolverse/src/mappings/smol-pets.ts:23]:
```suggestion
```

Remove commented code.

File: subgraphs/smolverse/tests/smol-bodies-gym/smol-bodies-gym.test.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  const collection = Collection.load(SMOL_BODIES_ADDRESS.toHexString()) as Collection;
+  assert.assertNotNull(collection);
+
+  // Assert token was created
+  const tokenId = `${collection.id}-0x1`;
+  const token = Token.load(tokenId) as Token;
+  assert.stringEquals(token.collection, collection.id);
+  assert.bigIntEquals(token.tokenId, BigInt.fromI32(1));
+
+  // Assert staked token was crerated
```
Comment (wyze) [inline subgraphs/smolverse/tests/smol-bodies-gym/smol-bodies-gym.test.ts:32]:
```suggestion
  // Assert staked token was created
```

File: subgraphs/smolverse/tests/smol-bodies-gym/smol-bodies-gym.test.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  assert.fieldEquals(STAKED_TOKEN_ENTITY_TYPE, stakedTokenId, "owner", USER_ADDRESS);
+});
+
+test("smol bodies dropping gym removes staked token", () => {
+  clearStore();
+
+  const joinEvent = createJoinGymEvent(USER_ADDRESS, 1);
+  handleJoinGym(joinEvent);
+
+  // Assert staked token was crerated
```
Comment (wyze) [inline subgraphs/smolverse/tests/smol-bodies-gym/smol-bodies-gym.test.ts:45]:
```suggestion
  // Assert staked token was created
```

**Review by wyze — CHANGES_REQUESTED** [review]
Mostly removing commented code and typos in comments.

File: subgraphs/smolverse/src/mappings/erc721.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import { log } from "@graphprotocol/graph-ts";
+
+import { ERC721, Transfer } from "../../generated/Smol Brains/ERC721";
+import { getIpfsJson, JSON } from "../helpers/json";
+import { updateTokenMetadata } from "../helpers/metadata";
+import { getOrCreateCollection, getOrCreateToken, getOrCreateUser } from "../helpers/models";
+import { isMint } from "../helpers/utils";
+
+// export function handleTransfer(event: Transfer, ipfsData: JSON | null = null): void {
```
[the remark this answers is not in the document]
Comment (alecananian) [reply subgraphs/smolverse/src/mappings/erc721.ts:9]: These commented-out function definitions allow you to run the tests locally until matchstick-as adds support for the `ipfs.cat` command. I can remove them if it's cluttering, or add some TODO comments to this effect (there's a TODO in the test file itself)

File: subgraphs/smolverse/src/mappings/erc721.ts (hunk 4 above)
Comment (alecananian) [reply subgraphs/smolverse/src/mappings/erc721.ts]: Fixed

**Review by wyze — APPROVED** [review]

**Pull request merged**